insights/specs/default.py

```diff
@@ -528,6 +528,28 @@ def package_and_java(broker):
     rpm_V_packages = simple_command("/usr/bin/rpm -V coreutils procps procps-ng shadow-utils passwd sudo")
     rsyslog_conf = simple_file("/etc/rsyslog.conf")
     samba = simple_file("/etc/samba/smb.conf")
+    saphostctl_listinstances = simple_command("/usr/sap/hostctrl/exe/saphostctrl -function ListInstances")
+
+    @datasource(saphostctl_listinstances, hostname, context=HostContext)
+    def sap_hana_sid(broker):
+        """
+        Command: Get the SID for running "HDB version".
+
+        Typical output of saphostctl_listinstances::
+        # /usr/sap/hostctrl/exe/saphostctrl -function ListInstances
+        Inst Info : SR1 - 01 - liuxc-rhel7-hana-ent - 749, patch 418, changelist 1816226
+
+        """
+        hana_ins = broker[DefaultSpecs.saphostctl_listinstances].content
+        hn = broker[DefaultSpecs.hostname].content[0].split('.')[0].strip()
+        results = set()
+        for ins in hana_ins:
+            ins_splits = ins.split(' - ')
+            if ins_splits[2].strip() == hn:
+                results.add(ins_splits[0].split()[-1].lower())
+        return list(results)
+
+    sap_hdb_version = foreach_execute(sap_hana_sid, "/bin/su %sadm -lc 'HDB version'", keep_rc=True)
```

Comment:
@kylape this has changed from the previous commit, but is still using the `su` command for a list of users that is being generated by the datasource.  Do we want to allow use of the `su` command?

Reply:
@kylape - Would you please to take a look at this `datasource`? Is the `su` command allowed?

Reply:
Yes, `sudo` works well too. I changed to `sudo` now, please check it.